bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusVMBELOHandler.java

```diff
@@ -12,27 +12,106 @@
  */
 package org.openhab.binding.velbus.internal.handler;
 
-import static org.openhab.binding.velbus.internal.VelbusBindingConstants.THING_TYPE_VMBELO;
+import static org.openhab.binding.velbus.internal.VelbusBindingConstants.*;
 
 import java.util.Arrays;
 import java.util.HashSet;
 import java.util.Set;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.velbus.internal.packets.VelbusPacket;
+import org.openhab.binding.velbus.internal.packets.VelbusRelayPacket;
+import org.openhab.binding.velbus.internal.packets.VelbusStatusRequestPacket;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.ChannelUID;
 import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
 import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
 
 /**
  * The {@link VelbusVMBELOHandler} is responsible for handling commands, which are
  * sent to one of the channels.
  *
  * @author Cedric Boon - Initial contribution
+ * @author Daniel Rosengarten - Add output support
  */
 @NonNullByDefault
 public class VelbusVMBELOHandler extends VelbusMemoHandler {
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(Arrays.asList(THING_TYPE_VMBELO));
 
+    private volatile boolean disposed = true;
+
+    private final ChannelUID outputChannel = new ChannelUID(thing.getUID(), CHANNEL_GROUP_OUTPUT, CHANNEL_OUTPUT);
+
     public VelbusVMBELOHandler(Thing thing) {
         super(thing);
     }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        disposed = false;
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+
+        disposed = true;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+
+        VelbusBridgeHandler velbusBridgeHandler = getVelbusBridgeHandler();
+        if (velbusBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            VelbusStatusRequestPacket packet = new VelbusStatusRequestPacket(getModuleAddress().getAddress());
+
+            byte[] packetBytes = packet.getBytes();
+            velbusBridgeHandler.sendPacket(packetBytes);
+        } else if (command instanceof OnOffType) {
+            byte commandByte = determineCommandByte((OnOffType) command);
+
+            VelbusRelayPacket packet = new VelbusRelayPacket(getModuleAddress(), commandByte);
+
+            byte[] packetBytes = packet.getBytes();
+            velbusBridgeHandler.sendPacket(packetBytes);
```

Comment:
Do I understand correctly this is intended to allow controlling the the output channel CH42 that's available on this module? This ends up sending a `COMMAND_SWITCH_RELAY_ON` to channel `ALL_CHANNELS` though, doesn’t it? Wouldn’t it be better to both send the message to specifically the output channel, and also to have users to use the channel properly, rather than skipping some steps like this?

Reply:
Here is the description of the packets that are sent to the module to change the state of the output :
**Switch open collector output off :**
```
SID10-SID9 = 00 (highest priority)
SID8...SID1 = Module address
RTR=0
DLC3...DLC0 = 2 data bytes received
DATABYTE1 = COMMAND_SWITCH_RELAY_OFF (0x01)
DATABYTE2 = channel bit = don't care
```
**Switch open collector output on :**
```
SID10-SID9 = 00 (highest priority)
SID8...SID1 = Module address
RTR=0
DLC3...DLC0 = 2 data bytes received
DATABYTE1 = COMMAND_SWITCH_RELAY_ON (0x02)
DATABYTE2 = channel bit = don't care
```
We could create a constant `public static final byte DONT_CARE = (byte) 0x00;` and use it instead of `ALL_CHANNELS`, but I don't think it's really a huge improvement.